src/config_store.c

```diff
@@ -10,14 +10,18 @@
 #include <sys/stat.h>
 #include <sys/statvfs.h>
 #include <sys/types.h>
+#include <sys/dir.h>
 #include <unistd.h>
+#include <dirent.h>
+#include <string.h>
 
 static char *AppendString(const char *front, const char *back)
 {
     size_t front_len = strlen(front);
     size_t back_len = strlen(back);
     char *dst = malloc(front_len + back_len + sizeof('\0'));
-    if (dst != NULL) {
+    if (dst != NULL)
```

Comment:
Why is so much of the code reformatted?  It'd probably be good to separate reformatting from a functional patch.

Reply:
We need a clang-format file for this repository to avoid all the reformatting.

@hardiktechnoplanet , can you add a file named `.clang-format` at the repository's root with the content below? VS Code uses it to apply the auto-format style.

```
---
#
# We start with the Google Style Guide as a basis for our style, and tweak
# as appropriate.
#
BasedOnStyle: Google
IndentWidth: 4
---
Language: Cpp
AccessModifierOffset: -4
AllowShortFunctionsOnASingleLine: Inline
AllowShortIfStatementsOnASingleLine: false
AllowShortLoopsOnASingleLine: false
BreakBeforeBraces: WebKit
ColumnLimit: 100
DerivePointerAlignment: false
KeepEmptyLinesAtTheStartOfBlocks: true
PointerAlignment: Right
SortIncludes: false
SpacesBeforeTrailingComments: 1
IndentCaseLabels: false
```